File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.cpp
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> arr) 
{
    vector<int> answer;
    int visited[10] = {0,};
    for (int i=0; i<arr.size(); i++){
        if (i==0)answer.push_back(arr[i]);
        else{
            if(arr[i] == arr[i-1]) continue;
            else answer.push_back(arr[i]);
        }
    }
    
    return answer;
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        ArrayList<Integer> answer = new ArrayList<Integer>();
        
        for (int i=0; i<arr.length; i++){
            if (i == 0) answer.add(arr[0]);
            else {
                if (arr[i] == arr[i-1])continue;
                else {
                    answer.add(arr[i]);
                }
            }
        }
        
        return answer.stream().mapToInt(i->i).toArray();
    }
}